src/System.Runtime.Extensions/src/System/AppDomain.cs

```diff
@@ -395,7 +395,8 @@ public IPrincipal GetThreadPrincipal()
                         if (s_getUnauthenticatedPrincipal == null)
                         {
                             Type type = Type.GetType("System.Security.Principal.GenericPrincipal, System.Security.Claims", throwOnError: true);
-                            Volatile.Write(ref s_getUnauthenticatedPrincipal, (Func<IPrincipal>)Delegate.CreateDelegate(typeof(Func<IPrincipal>), type, "GetDefaultInstance"));
+                            Volatile.Write(ref s_getUnauthenticatedPrincipal,
+                                (Func<IPrincipal>)Delegate.CreateDelegate(typeof(Func<IPrincipal>), type, "GetDefaultInstance", ignoreCase: false, throwOnBindFailure: false));
```

Comment:
I guess the code did not throw PNSE and just let the reflection throw the missing method exception because of `UnauthenticatedPrincipal` should be present on all platforms. If it is missing, it means that the assembly was trimmed incorrectly or something similar.

Reply:
I would leave it as is. Changing it to throw PNSE would not be an improvement.

It may be worth adding a comment.

Reply:
Makes sense. Will do.